core/sequencer/server.go

```diff
@@ -545,25 +550,33 @@ func (s *Server) HighWatermarks(ctx context.Context, directoryID string, lastMet
 	}
 	// TODO(gbelvin): Get HighWatermarks in parallel.
 	for _, logID := range logIDs {
-		low := time.Unix(0, ends[logID])
+		low, ok := ends[logID]
+		if !ok {
+			// Here be dragons.
+			// When getting the watermark for a new log that is not in the lastMeta set,
+			// we need to pick a lower bound starting point for reading.  time.Time{} is
```

Comment:
This comment may be better placed on the zero value itself rather than here. Your call.